checker/src/main/java/org/checkerframework/checker/mustcall/MustCallVisitor.java

```diff
@@ -0,0 +1,193 @@
+   * that does not include "close" is substituted.
+   */
+  @Override
+  protected void commonAssignmentCheck(
+      AnnotatedTypeMirror varType,
+      AnnotatedTypeMirror valueType,
+      Tree valueTree,
+      @CompilerMessageKey String errorKey,
+      Object... extraArgs) {
+    if (Arrays.asList(extraArgs).contains(ElementKind.RESOURCE_VARIABLE)) {
+      AnnotationMirror varAnno = varType.getAnnotationInHierarchy(atypeFactory.TOP);
+      AnnotationMirror valAnno = valueType.getAnnotationInHierarchy(atypeFactory.TOP);
+      if (atypeFactory
+          .getQualifierHierarchy()
+          .isSubtype(atypeFactory.withoutClose(valAnno), atypeFactory.withoutClose(varAnno))) {
+        return;
+      }
+    }
+    super.commonAssignmentCheck(varType, valueType, valueTree, errorKey, extraArgs);
+  }
+
+  /**
+   * This method typically issues a warning if the result type of the constructor is not top,
+   * because in top-default type systems that indicates a potential problem. The must call checker
+   * does not need this warning, because it expects the type of all constructors to be {@code
+   * MustCall({})} (by default) or some other {@code MustCall} type, not the top type.
+   *
+   * <p>Instead, this method checks that the result type of a constructor is a supertype of the
+   * declared type on the class, if one exists.
+   *
+   * @param constructorType AnnotatedExecutableType for the constructor
+   * @param constructorElement element that declares the constructor
+   */
+  @Override
+  protected void checkConstructorResult(
+      AnnotatedExecutableType constructorType, ExecutableElement constructorElement) {
+    AnnotatedTypeMirror defaultType =
+        atypeFactory.getAnnotatedType(ElementUtils.enclosingTypeElement(constructorElement));
+    AnnotationMirror defaultAnno = defaultType.getAnnotationInHierarchy(atypeFactory.TOP);
+    AnnotationMirror resultAnno =
+        constructorType.getReturnType().getAnnotationInHierarchy(atypeFactory.TOP);
+    if (!atypeFactory.getQualifierHierarchy().isSubtype(defaultAnno, resultAnno)) {
+      checker.reportError(
+          constructorElement, "inconsistent.constructor.type", resultAnno, defaultAnno);
+    }
+  }
+
+  /**
+   * Change the default for exception parameter lower bounds to bottom (the default), to prevent
+   * false positives. This is unsound; see the discussion on
+   * https://github.com/typetools/checker-framework/issues/3839. TODO: change checking of throws
+   * clauses to require that the thrown exception is @MustCall({}). This would probably eliminate
+   * most of the same false positives, without adding undue false positives.
+   *
+   * @return a set containing only the @MustCall({}) annotation
+   */
+  @Override
+  protected Set<? extends AnnotationMirror> getExceptionParameterLowerBoundAnnotations() {
+    return Collections.singleton(atypeFactory.BOTTOM);
+  }
+
+  /**
+   * Does not issue any warnings.
```

Comment:
The (linked) explanation at https://github.com/typetools/checker-framework/issues/3178 makes it clear that it is a correctness issue - without this change, unnecessary false positives are issued at every use of an annotation with arguments in checked source code. This override exists to prevent those false positives, as the third paragraph of this Javadoc comment states.

I'm not sure how I could make this comment more clear, since it already contained the statement that the override is necessary to avoid false positives, and linked to the Checker Framework issue that makes it necessary.

Reply:
I don't like comments that link to a discussion elsewhere.
 * The code is no longer self-contained.
 * The discussion elsewhere is often much longer and less focused on this particular piece of code.  That is, it is unnecessarily difficult to understand, for someone coming from the code.

Could you write a sentence like the one in this comment?